x-pack/plugins/observability_ai_assistant/server/plugin.ts

```diff
@@ -105,10 +105,34 @@ export class ObservabilityAIAssistantPlugin
       };
     }) as ObservabilityAIAssistantRouteHandlerResources['plugins'];
 
+    const getModelId = once(async () => {
+      // Using once to make sure the same model ID is used during service init and Knowledge base setup
+
+      try {
+        // Wait for the ML plugin's dependency on the internal saved objects client to be ready
+        const [_, pluginsStart] = await core.getStartServices();
+
+        // Wait for the license to be available so the ML plugin's guards pass once we ask for ELSER stats
+        await pluginsStart.licensing.refresh();
```

Comment:
curious why we need this and not use lastValueFrom(license$)? Does this trigger some kind of HTTP call?

Reply:
It does trigger a HTTP call to get the license information, I think I just used this because I only wanted to express "wait for license information to be available". 

I suppose we could swap this to a subscribe on waiting for lastValueFrom to get any value?

Reply:
@miltonhultgren yeah, I would prefer something that doesn't trigger a call. but lastValueFrom(license$) might not appropriately wait until the license has been resolved. One alternative would be to wait until there's _a_ license, and then call the ML API.

Reply:
Fixed 👍🏼